src/App.tsx

```diff
@@ -1,14 +1,85 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 import { Loader } from './components/Loader';
+import { TodoModal } from './components/TodoModal';
+
+function getPreparedTodos(todos: Todo[], query: string) {
+  let preparedTodos = [...todos];
+
+  if (query) {
+    const prepQuery = query.trim().toLowerCase();
+
+    preparedTodos = preparedTodos.filter(
+      todo => todo.title.toLowerCase().includes(prepQuery),
+    );
+  }
+
+  return preparedTodos;
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const visibleTodos = getPreparedTodos(todos, query);
+  const [loading, setLoading] = useState(true);
+  const [modalVisible, setModalVisible] = useState(true);
```

Comment:
you don't need this state
you can check `selectedTodo` variable to know if the modal should be opened or not - if `selectedTodo` is null - do not show modal, if there is `Todo` - show the modal

Reply:
Not fixed